build.zig

```diff
@@ -148,6 +196,33 @@ pub fn build(b: *std.Build) void {
         lib.addConfigHeader(revision_header);
         lib.installConfigHeader(revision_header);
     }
+
+    const use_hidapi = b.option(bool, "use_hidapi", "Use hidapi shared library") orelse target.result.isAndroid();
+
+    if (use_hidapi) {
+        const hidapi_lib = b.addSharedLibrary(.{
+            .name = "hidapi",
+            .target = target,
+            .optimize = optimize,
+            .link_libc = true,
+        });
+        hidapi_lib.addIncludePath(sdl_include_path);
+        hidapi_lib.addIncludePath(b.path("include-pregen"));
+        hidapi_lib.root_module.addCSourceFiles(.{
+            .root = b.path(""),
+            .files = &[_][]const u8{
+                "src/hidapi/android/hid.cpp",
+            },
+            .flags = &.{"-std=c++11"},
+        });
+        hidapi_lib.linkSystemLibrary("log");
+        hidapi_lib.linkLibCpp();
+        lib.linkLibrary(hidapi_lib);
+        b.installArtifact(hidapi_lib);
+    } else {
+        lib.root_module.addCMacro("SDL_HIDAPI_DISABLED", "");
```

Comment:
While I don't know if this will have a negative effect, I'm wary of having this defined (by default) for all other targets. I don't see that happening in upstream's build scripts, and it looks like some of the prefab target config headers already have it covered. It might be better to remove this else branch for now.

Reply:
In my version, I just always included `hidapi` as on by default.
https://github.com/silbinarywolf/zig-android-sdk/blob/7133284eb26af62b900fefbc451186edbf997216/examples/sdl2/third-party/sdl2/build.zig#L121-L123

Reply:
Makes sense, removed it.